OWML.ModHelper/ModInputHandler.cs

```diff
@@ -0,0 +1,447 @@
+				{
+					curval -= 1f * axisdir;
+					//DebugInput.logger.Log("succesfully ignored " + pos.ToString());
+				}
+				if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+				{
+					curval += 1f * axisdir;
+					//DebugInput.logger.Log("succesfully ignored " + neg.ToString());
+				}
+				val.SetValue(__instance, curval);
+			}
+
+			public static void DAxisUpdPost(DoubleAxisCommand __instance)
+			{
+				ModInputHandler._self.RegisterGamesBinding(__instance);
+				if (!OWInput.UsingGamepad())
+				{
+					KeyCode pos, neg;
+					FieldInfo val = typeof(DoubleAxisCommand).GetField("_value", BindingFlags.NonPublic | BindingFlags.Instance);
+					Vector2 curval = (Vector2)(val.GetValue(__instance));
+					pos = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardXPos", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					neg = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardXNeg", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					if (Input.GetKey(pos) && ModInputHandler._self.ShouldIgnore(pos))
+					{
+						curval.x -= 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + pos.ToString());
+					}
+					if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+					{
+						curval.x += 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + neg.ToString());
+					}
+					pos = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardYPos", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					neg = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardYNeg", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					if (Input.GetKey(pos) && ModInputHandler._self.ShouldIgnore(pos))
+					{
+						curval.y -= 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + pos.ToString());
+					}
+					if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+					{
+						curval.y += 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + neg.ToString());
+					}
+					val.SetValue(__instance, curval);
+				}
+			}
+		}
+
+		public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+		{
+			_console = console;
+			_logger = logger;
+			patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), "SAxisRemPre");
+			patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), "DAxisRemPre");
+			patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), "SAxisUpdPost");
+			patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), "DAxisUpdPost");
+			_self = this;
+		}
+
+		private void UpdateCombo()
+		{
+			if (Time.realtimeSinceStartup - lastUpdate > 0.01f)
+			{
+				lastUpdate = Time.realtimeSinceStartup;
+				Int64 hash = 0;
+				int[] keys = new int[7];
+				int t = 0;
+				bool countdowntrigger = false;
+				for (int code = 8; code < 350; code++)
```

Comment:
Avoid magic numbers. Why 8? Why 350? Give then names by creating constants. Same for all magic numbers in this file. If you find that making constants is unnecessary for some of them, at least add a small comment.

Reply:
done